pkg/db/db.go

```diff
@@ -7,27 +7,31 @@ Copyright 2018
 Licensed under the Apache License, Version 2.0 (the "License")
 
 */
-package utils
+package db
 
 import (
 	"database/sql"
-	_ "github.com/lib/pq"
 	"fmt"
 	"time"
-	"github.com/pivotal-gss/gpmt2/log"
+
+	_ "github.com/lib/pq"
+	log "github.com/sirupsen/logrus"
 )
 
-var _db *sql.DB
+// globally defined, because the db object/connection should generally be in go
+// FIXME: Should this be exported?
+var db *sql.DB
```

Comment:
*sql.DB is thread-safe and therefore can be accessed concurrently, if you define it within a function you are basically starting a pointer and then passing a pointer , which doesn't make a lot of sense.

So having this defined globally can help in every sub function to know the connection pointer and doesn't chew up memory ..